lib/build.rb

```diff
@@ -175,12 +176,18 @@ def fetch_cmd
       cmd = []
       cmd << "cd #{project_dir}"
       cmd << "git reset --hard"
-      cmd << "git clean -fdx"
+      cmd << "git clean #{git_clean_opts}"
       cmd << "git remote set-url origin #{@repo_url}"
       cmd << "git fetch origin"
       cmd.join(" && ")
     end
 
+    def git_clean_opts
+      opts = @cache_pattern_list.collect { |item| "--exclude='#{item}'"}
+      opts.unshift("-fdx")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.